scripts/commands/convertApiDocsToHistorical.ts

```diff
@@ -0,0 +1,165 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { $ } from "zx";
+import { zxMain } from "../lib/zx";
+import { globby } from "globby";
+import { pathExists, getRoot } from "../lib/fs";
+import { readFile, writeFile } from "fs/promises";
+import { mkdirp } from "mkdirp";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import transformLinks from "transform-markdown-links";
+
+interface Arguments {
+  [x: string]: unknown;
+  package: string;
+}
+
+const PACKAGES: string[] = [
+  "qiskit-ibm-runtime",
+  "qiskit-ibm-provider",
+  "qiskit",
+];
+
+const readArgs = (): Arguments => {
+  return yargs(hideBin(process.argv))
+    .version(false)
+    .option("package", {
+      alias: "p",
+      type: "string",
+      choices: PACKAGES,
+      demandOption: true,
+      description: "Which package to convert",
+    })
+    .parseSync();
+};
+
+zxMain(async () => {
+  const args = readArgs();
+
+  const pkgName = PACKAGES.find((pkgName) => pkgName === args.package);
+  if (pkgName === undefined) {
+    throw new Error(`Unrecognized package: ${args.package}`);
+  }
+
+  const packageFile = await readFile(
+    `${getRoot()}/docs/api/${pkgName}/_package.json`,
+    { encoding: "utf8" },
+  );
+  const packageInfo = JSON.parse(packageFile);
+  const versionMatch = packageInfo.version.match(/^(\d+\.\d+)/);
+  const versionWithoutPatch = versionMatch[0];
+
+  const projectNewHistoricalFolder = `${getRoot()}/docs/api/${pkgName}/${versionWithoutPatch}`;
+  if (await pathExists(projectNewHistoricalFolder)) {
+    console.error(
+      `${pkgName} has already a historical version ${versionWithoutPatch}`,
+    );
+    process.exit(1);
+  }
+
+  await mkdirp(projectNewHistoricalFolder);
+  await mkdirp(`${projectNewHistoricalFolder}/apidoc`);
+  await mkdirp(`${projectNewHistoricalFolder}/stubs`);
+
+  copyApiDocs(pkgName, versionWithoutPatch);
+  generateJsonFiles(
+    pkgName,
+    packageInfo.version,
+    versionWithoutPatch,
+    projectNewHistoricalFolder,
+  );
+  copyImages(pkgName, versionWithoutPatch);
+
+  if (pkgName == "qiskit") {
+    await mkdirp(`${projectNewHistoricalFolder}/release-notes`);
+    copyReleaseNotes(pkgName, versionWithoutPatch);
```

Comment:
If it's not qiskit, it should still copy the release notes. Only a file rather than folder. You could probably keep one helper function `copyReleaseNotes` and move this conditional logic into there.

Reply:
Currently, the script is also copying the release notes for other projects, but given that we don't have any subfolder, those release notes are treated as every other doc and copied in the `copyApiDocsAndUpdateLinks` function.

It's true that the name `copyReleaseNotes` can be confusing, so I changed it to `copyReleaseNotesFolder` to make explicit that we only copy the folder if it exists.